core/chains/evm/headtracker/types/types.go

```diff
@@ -15,8 +15,8 @@ type HeadSaver interface {
 	// Save updates the latest block number, if indeed the latest, and persists
 	// this number in case of reboot.
 	Save(ctx context.Context, head *evmtypes.Head) error
-	// LoadFromDB loads latest EvmHeadTrackerHistoryDepth heads, returns the latest chain.
-	LoadFromDB(ctx context.Context) (*evmtypes.Head, error)
+	// Load loads latest EvmHeadTrackerHistoryDepth heads from the DB, returns the latest chain.
+	Load(ctx context.Context) (*evmtypes.Head, error)
 	// LatestHeadFromDB returns the highest seen head from DB.
 	LatestHeadFromDB(ctx context.Context) (*evmtypes.Head, error)
```

Comment:
should we also generalize the naming of `LatestHeadFromDB`? if we do, what's the difference with `LatestChain`? or can these two methods be folded together?

Reply:
I think that we can leave generalising `LatestHeadFromDB` naming out for now, as it is not used by the Headtracker? 

LatestHeadFromDB fetches the latest head stored from the DB:
https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/headtracker/head_saver.go#L51

LatestChain is similar where it returns the block header with the highest number that has been seen. But it abstracts away how the latest Head is being fetched.

Reply:
`LatestChain` is the method being called by the Headtracker and it calls `LatestHeadFromDB`

Reply:
`LatestHeadFromDB` fetches a single DB entry for the log broadcaster, which might not even be needed. I wouldn't generalize it.